server/config/config.go

```diff
@@ -34,15 +38,44 @@ type BaseConfig struct {
 
 	// InterBlockCache enables inter-block caching.
 	InterBlockCache bool `mapstructure:"inter-block-cache"`
+}
 
-	Pruning              string `mapstructure:"pruning"`
-	PruningKeepEvery     string `mapstructure:"pruning-keep-every"`
-	PruningSnapshotEvery string `mapstructure:"pruning-snapshot-every"`
+// APIConfig defines the API listener configuration.
+type APIConfig struct {
+	// Enable defines if the API server should be enabled.
+	Enable bool `mapstructure:"enable"`
```

Comment:
@alexanderbez just saw this is not used, is that correct?

Reply:
WDYM not used? It used.

Reply:
Sorry, maybe I am wrong but inside here

https://github.com/cosmos/cosmos-sdk/blob/8ee7d1f403cc9eda793348deb721c53f34b3f7fe/server/api/server.go#L48

I dont see any if Config.API.Enabled and I am going crazy with it!

Reply:
https://github.com/cosmos/cosmos-sdk/blob/master/server/start.go#L195

We should probably create the config before that and use that instead. Mind creating a small PR/

Reply:
Leave it in my hands!